tests/mock_flags.py

```diff
@@ -24,43 +22,68 @@
 FLAGS = flags.FLAGS
 
 
+class _MockFlag(object):
+  """Mock version of a Flag object."""
+
+  def __init__(self):
+    self.present = False
+    self.validators = ()
```

Comment:
Yes, by gflags: https://github.com/gflags/python-gflags/blob/master/gflags.py#L1080
We substitute the real FlagValues instance's internal flag dict with our _MockFlagDict, so that the scattered references to FLAGS in every module will all be affected. I'll expand this class's docstring a bit to cover the attributes.